src/daemon/daemon.cpp

```diff
@@ -2518,23 +2518,13 @@ try
         auto spec_it = vm_instance_specs.find(instance_name);
         assert(spec_it != vm_instance_specs.end() && "missing instance specs");
 
-        // TODO@snapshots need a way to query the existence of a snapshot before consuming it
-        try
-        {
-            auto snapshot = vm_ptr->get_snapshot(request->snapshot());
-        }
-        catch (const std::out_of_range&)
-        {
-            return status_promise->set_value(
-                grpc::Status{grpc::NOT_FOUND, fmt::format("snapshot \"{}\" does not exist", request->snapshot())});
-        }
+        vm_ptr->get_snapshot(request->snapshot());
```

Comment:
You removed the exception handing? I thought `NOT_FOUND` was appropriate. You can also place it together with the other catch at the end of the function. I would also add a comment to clarify why this is here.